main/src/library/Object.flix

```diff
@@ -20,7 +20,21 @@ namespace Object {
     ///
     pub def isNull(x: a): Bool =
         import static java.util.Objects.isNull(##java.lang.Object): Bool \ {};
-        isNull(x as ##java.lang.Object)
+        match reifyType a {
+            case ReifiedType.ReifiedUnit => false
+            case ReifiedType.ReifiedBool => false
+            case ReifiedType.ReifiedChar => false
+            case ReifiedType.ReifiedFloat32 => false
+            case ReifiedType.ReifiedFloat64 => false
+            case ReifiedType.ReifiedInt8 => false
+            case ReifiedType.ReifiedInt16 => false
+            case ReifiedType.ReifiedInt32 => false
+            case ReifiedType.ReifiedInt64 => false
+            case ReifiedType.ReifiedBigInt => isNull(x as ##java.lang.Object)
```

Comment:
Does this work :O I though it would break when isNull is called with cat2 types

Reply:
We need to fix (something) so that we eliminate all the other cases.

Reply:
Perhaps we need a dedicated pattern match construct?

Reply:
It doesn't work :D

Reply:
I tried something with it but it ended up needing GADTs...

Reply:
Hardcoded into the compiler which monomorph guarantees to remove.

Reply:
I think for now it might work by eta-expanding the isNull function and casting it
```
((y -> isNull(y)) as a -> Bool)(x)
```

Reply:
Jeezus. We should fix this instead!